src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -0,0 +1,136 @@
+package uk.gov.dwp.dataworks.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.awscore.AwsRequest
+import software.amazon.awssdk.core.SdkRequest
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.ecs.EcsClient
+import software.amazon.awssdk.services.ecs.model.*
+import software.amazon.awssdk.services.ecs.model.LoadBalancer
+import software.amazon.awssdk.services.elasticloadbalancingv2.ElasticLoadBalancingV2Client
+import software.amazon.awssdk.services.elasticloadbalancingv2.model.*
+
+import java.net.http.HttpRequest
+
+@Service
+class TaskDeploymentService {
+    @Autowired
+    lateinit var credentialsService: CredentialsService
+
+    val awsRegion: Region = kotlin.runCatching { Region.of(System.getenv(ConfigKey.AWS_REGION.toString()))}.getOrDefault(Region.EU_WEST_2)
+
+    private fun createService (ecs_cluster_name: String, user_name: String, ecsClient: EcsClient, targetGroupArn: String) {
+
+        val alb: LoadBalancer = LoadBalancer.builder().targetGroupArn(targetGroupArn).containerPort(443).build()
+
+        val serviceBuilder = CreateServiceRequest.builder().cluster(ecs_cluster_name).loadBalancers(alb).serviceName("${user_name}_test").taskDefinition("mhf_sample_task").loadBalancers(alb).desiredCount(1).build()
+        println("Creating Service...")
+
+        try {
+            val service = ecsClient.createService(serviceBuilder)
+            println("service.responseMetadata = ${service.responseMetadata()}")
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
+    }
+
+    fun getPriorityVal (rulesResponse : DescribeRulesResponse) : Int {
+
+        var consecVals : Int = 1
+        for (i in rulesResponse.rules()) {
+            if (i.priority() != "default") {
+                var intVal = Integer.parseInt(i.priority())
+                if (intVal == consecVals) consecVals = consecVals + 1
+            }
+        }
+        if (consecVals >= 1000) throw Exception("The upper limit of 1000 rules has been reached for this load balancer.")
+        return consecVals;
+    }
```

Comment:
I feel as if the logic here is not strict enough. I would prefer to see some logic here that takes into account the whole set of rules and find one which is not present. Also, some explanation as to what is happening in this method and better named variables would really help with the understanding of this code.

Reply:
returned values are ordered by priority so this seems the simplest way to get the lowest available slot and both methods work in similar way

Reply:
I agree that the logic here is correct, however something that I can't put my finger on seems off about this code. I know that's a rubbish comment but I do have a couple issues I can point out:
- We always are iterating O(n) times here, where n is the number of rules on the ALB. This wastes CPU cycles, as it doesn't matter if the lowest vacant value is 1 or 999.
- The `for` loop is a little misconstrued - instead of iterating over a finite number of integers (1000) and breaking at the first opportunity, we are iterating over an unknown number of values and only breaking once all of them are processed. As we have no control over the value of `n` we could introduce issues here.